### ISSUE TITLE
Add comment to likes 

### ISSUE THREAD

**Issue by colin-walker** [EdwardHinkle/indigenous-ios#33]
Just as we will be able to add content to a reply how about giving the option to do the same for likes rather than limiting it to the action.

**Comment by EdwardHinkle** [comment]
That's a great question, @colin-walker.

When selecting the like option, that is using micropub/microformats2's `like-of` property. The standard interpretation of like-of posts is that it is just an h-entry with a `like-of` property but no content. This means when most websites receive a webmention notification of a like, anything except the existence of a like will disappear.

There's some more information over here: https://indieweb.org/like#mark_up_and_post_a_like

At least that's how I understand it. I'm not sure if @aaronpk has any additional thoughts about that.

**Comment by aaronpk** [comment]
Yep! Likes by definition have no other content. As soon as there is some user-entered content, it becomes a reply. 

Now that said, there are some interesting UI opportunities here, along the lines of how Micro.blog will add the post name field if you type a note over a certain length. I think that's beyond the scope of a like, but I could be convinced otherwise. The analogous progressive enhancement I'm imagining would be if you click the "like" option and then start typing some content, it changes from a "like" to maybe a "bookmark" with content. I struggle to think of a way that would be easily explained though.

**Comment by colin-walker** [comment]
I looked at the indieweb wiki and saw the example of a like with a nested h-cite allowing you to display a quote from the liked site so figured it might be an option.

My own implementation allows for the addition of extra body content to the post. Yes, the like is just the action and that’s all the target site sees but I think it’s nice to say *why* I liked it.

**Comment by colin-walker** [comment]
Here’s an example of a like on my blog:
https://colinwalker.blog/2017/11/13/13112017-1406/

I don’t know if it would be possible to replicate something like that via micropub. Or does the nature of a like negate the posting of any additional text?

**Comment by EdwardHinkle** [comment]
I think there are a couple components to this.

It is technologically possible for me to add a config option hidden in an advanced area that would let people add commentary related to why they are liking something, and that could probably be sent as a `content` attribute through micropub. This would prevent people from accidentally stumbling into having a commentary spot for a like, while allowing people who like to provide that on their site.

I think the question is how your micropub server would respond to that. I'm not sure if you're using a custom plugin or one of the community IndieWeb plugins. Let me think through this a bit more to make sure there's not some dire mistake I'm missing ;) But I'm open to this idea, and I'm adding replies to the next build, so it's a good time to think this through because if so I would use much of the same code as the reply feature for the "like commentary" feature

**Comment by colin-walker** [comment]
It’s the standard indieweb one: https://wordpress.org/plugins/micropub/

Thanks for considering it :)
